portfolio/src/main/java/com/google/sps/data/BaseTaskScheduler.java

```diff
@@ -0,0 +1,32 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.data;
+
+import java.time.Instant;
+import java.util.Collection;
+
+// TODO: Add test coverage for scheduling algorithms
+public abstract class BaseTaskScheduler {
```

Comment:
I think there should be an abstract method that returns the algorithm type.

abstract SchedulingAlgorithmType getSchedulingAlgorithmType();

And classes that implement the abstract class can override that method.

Reply:
Done.